gpslogger/src/main/java/com/mendhak/gpslogger/ui/fragments/settings/LoggingSettingsFragment.java

```diff
@@ -166,12 +169,12 @@ public boolean onPreferenceChange(final Preference preference, Object newValue)
             }
             catch(Exception e){
                 LOG.error("Could not create chosen directory path", e);
-                Dialogs.error(getString(R.string.error), getString(R.string.pref_logging_file_no_permissions), e.getMessage(), e, getActivity());
+                displayFileAccessError();
```

Comment:
Note: during my tests, only the error below with `isAllowedToWriteTo` showed. I don't know if we really need to change this.

Reply:
I did some testing and ran into both situations.  One with the directory not being created.  The other was `isAllowedToWriteTo` was returning _true_ but when the app tried to create a file later, it failed.  So I've added a very deliberate test here to create a file then delete it.